src/main/java/io/github/bobocodebreskul/context/scan/ClassPathAnnotatedBeanScanner.java

```diff
@@ -0,0 +1,83 @@
+package io.github.bobocodebreskul.context.scan;
+
+import java.util.*;
+
+import static java.util.Arrays.asList;
+import static java.util.function.Predicate.not;
+import static java.util.stream.Collectors.toSet;
+
+/**
+ * Class to recursively scan packages and search for bean definition classes. Additionally, performs package scan of
+ * all found configurations.
+ * Return found bean definitions in all package tree.
+ *
+ * @author Vitalii Katkov
+ */
+public class ClassPathAnnotatedBeanScanner {
+
+    private static final String PACKAGE_DELIMITER = ".";
```

Comment:
Let's have separate class with constants

Reply:
I would move only `EMPTY` constant to some general constants class, `PACKAGE_DELIMITEER` is too specific to be available in it.